### ISSUE TITLE
GroupACL: No effect when assigned to user

### ISSUE THREAD

**Issue by ptr1120** [nijel/weblate#1024]
Hello,

thanks for the new version of weblate. I have just upgraded from 2.4->2.5 and everything worked fine, except the eagerly awaited language-based permissions feature.

Environment: I have 2 Projects with multiple Components for .net Ressource files. I have a "testuser" that has 3 roles ("Users", "Project1_role", "Project2_role") that enable this user to access these projects and edit languages.
I want to create a "GroupACL" that globally (for all Projects) locks the e.g. japanese language to users of a role named "Japanese_translators".

### Steps to reproduce
1. Create a role "Japanese_translators" with no further permission preferences
2. Create a  "GroupACL" for the role "Japanese_translators" with the language "Japanese" (I assigned here no project / component, since I wan't to lock the language globally)
=> Effect: Now the "Japanese" language is not editable in both projects for all users => OK
3. Assign the role "Japanese_translators" to the user "testuser" -> Login with "testuser"
=> Effect:The user still is not able to edit the japanese language.

### Expected behaviour
When I assign the GroupACL "Japanese_translators" to "testuser", the user should be able to edit japanese translations.

### Actual behaviour
The user can not edit japanese translations although he has the role  "Japanese_translators".

### Server configuration
 * Weblate 2.5
 * Python 2.7.3
 * Django 1.9.4
 * six 1.10.0
 * python-social-auth 0.2.14
 * Translate Toolkit 1.13.0
 * Whoosh 2.7.3
 * Git 1.7.10.4
 * Pillow (PIL) 1.1.7
 * dateutil 2.5.0
 * lxml 3.5.0
 * django-crispy-forms 1.6.0
 * compressor 1.6
 * pyuca N/A
 * pyLibravatar N/A
 * Mercurial 3.7.2
 * Database backends: django.db.backends.postgresql_psycopg2

**Comment by nijel** [comment]
You need to add desired permissions to the group. Just creating empty group without any permissions doesn't have effect.

**Comment by ptr1120** [comment]
But the user already has the groups "Users", "Project1_role", "Project2_role" and is able to edit translations of both projects. What are these desired permissions that are needed for a GroupACL?
When I add the permissions "trans | Project | can access Project1, ... Project2" to "Japanese_translators" this has no effect.

**Comment by nijel** [comment]
You need to add at least the edit translation privilege to the group. The Group ACLs work in a way that they add it's privleges if the group/project/component matches.

**Comment by ptr1120** [comment]
thank you but this does have no effect. My testuser still has no save button at japanese translation neither for "All strings" nor for "translated strings".

The testuser has the following roles and permissions:
* Users
    * trans | comment | Can add comment
    * trans | dictionary | Can add dictionary
    * trans | dictionary | Can change dictionary
    * trans | dictionary | Can delete dictionary
    * trans | dictionary | Can import dictionary
    * trans | Component | Can see VCS repository URL
    * trans | suggestion | Can accept suggestion
    * trans | suggestion | Can add suggestion
    * trans | suggestion | Can delete suggestion
    * trans | check | Can ignore check results
    * trans | suggestion | Can vote for suggestion
    * trans | translation | Can lock whole translation project
    * trans | translation | Can overwrite with translation upload
    * trans | translation | Can upload translation
    * trans | translation | Can use machine translation
    * trans | unit | Can save template
    * trans | unit | Can save translation
* "Project1_role"
   * trans | Project | Can access project Project1
* "Project2_role"
   * trans | Project | Can access project Project2
* "Japanese_translators"
  * trans | translation | Can change translation
 
Whereas "Japanese_translators" is assigned under Group ACL to the language "Japanese"

Any idea?

**Comment by nijel** [comment]
Not sure, maybe I'm missing something. @matejcik might have better ideas as he wrote the Group ACL stuff.

**Comment by matejcik** [comment]
i'm not sure which permissions you need to edit translations. "Can change translation" sounds about right.
Maybe the GroupACL is interfering with the project-level ACLs?
Please try disabling the project ACLs and see if that helps.
(if that helps, that is also some sort of a bug, but at least we'll know *what* sort)

I'll try to reproduce it on monday.

**Comment by ptr1120** [comment]
Hello,

thanks for your response. I removed the roles "Project1_role" and "Project2_role" from my testuser and added the permissions "trans | Project | Can access project Project1", "...Project2" directly to the "Japanese_translators". Now  the testuser is even not allowed to access any project ("You are not allowed to access project Project1.",...). The current roles and permissions of testuser are:

* Users
    * trans | comment | Can add comment
    * trans | dictionary | Can add dictionary
    * trans | dictionary | Can change dictionary
    * trans | dictionary | Can delete dictionary
    * trans | dictionary | Can import dictionary
    * trans | Component | Can see VCS repository URL
    * trans | suggestion | Can accept suggestion
    * trans | suggestion | Can add suggestion
    * trans | suggestion | Can delete suggestion
    * trans | check | Can ignore check results
    * trans | suggestion | Can vote for suggestion
    * trans | translation | Can lock whole translation project
    * trans | translation | Can overwrite with translation upload
    * trans | translation | Can upload translation
    * trans | translation | Can use machine translation
    * trans | unit | Can save template
    * trans | unit | Can save translation
* "Japanese_translators"
  * trans | translation | Can change translation
  * trans | Project | Can access project Project1
  * trans | Project | Can access project Project2


Best regards,
Peter

**Comment by ptr1120** [comment]
@matejcik: Any ideas what I can try to fix this on my side? Have you been able to reproduce this behavior?

**Comment by matejcik** [comment]
sorry, i got pretty seriously ill in the meantime

anyway, i was able to reproduce the problem.
adding "trans | unit | Can save translation" to "Japanese_translators" fixed it for me
so the GroupACL mechanism seems to work fine, but it is not completely clear *which* permissions you need on the selected group. maybe you will encounter more things you would need to add to Japanese_translators to make it work for you.

**Comment by TTmiro** [comment]
Hi folks,

I have a problem sounds quite similar to ptr1120: I use Weblate 2.7 on a similar configuration as ptr1120. 

I try to achieve the following:

I want to work with several translators, each translator should be able to save translations for his/her language but not for any other language.

I created the following groups

	translators_chinese
	translators_french
	translators_german
	translators_russian
	translators_spanish
	translators_vietnamese

with the following rights

trans | translation | Can change translation
trans | translation | Can save translation
trans | Project | Can access project Project Name

in Group ACL I created the following languages and linked it to the corresponding group

        Vietnamese	Project Name/messages	translators_vietnamese
	Spanish	Project Name/messages	   translators_spanish
	Russian	Project Name/messages	   translators_russian
	German	Project Name/messages	   translators_german
	French	Project Name/messages	   translators_french
	Chinese	Project Name/messages	   translators_chinese
	German	Project Name/messages	   translators_german

Finally I linked users to groups (e.g.):

user.name
       translators_french
       Users
       Project Name

By logging in as user.name I can access all languages (this is fine) without save or suggest button - but these buttons are as well missing for French. 

How can I enable these buttons for certain languages? 

Thank you for reading my question - your help his highly appreciated.

Kind regards,
TTmiro
